cibuildwheel/platforms/windows.py

```diff
@@ -370,15 +371,15 @@ def build(options: Options, tmp_path: Path) -> None:
                 version=config.version,
                 tmp_dir=identifier_tmp_dir,
             )
-            dependency_constraint_flags: Sequence[PathOrStr] = (
-                ["-c", constraints_path] if constraints_path else []
+            dependency_constraint_flags = (
+                ["-c", constraints_path.as_uri()] if constraints_path else []
             )
```

Comment:
```suggestion
            dependency_constraint_flags = constraint_flags(constraints_path)
```
but the same remark as macOS applies here so might not be worth merging the suggestion as-is